helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -1705,4 +1734,27 @@ private static void verifyToken(final String token, final Job job)
     }
   }
 
+  private List<TaskStatus.State> getPreviousJobStates(final JobId jobId, final String host) {
+    List<TaskStatus.State> previousStates;
```

Comment:
Seems reasonable, but IntelliJ complains that `previousStates = emptyList();` in the catch might have already been assigned to.